contracts/cw1-whitelist-ng/src/multitest/contract.rs

```diff
@@ -0,0 +1,239 @@
+use crate::msg::*;
+use crate::query::*;
+use crate::state::Cw1WhitelistContract;
+use anyhow::{bail, Result as AnyResult};
+use cosmwasm_std::{
+    Addr, Binary, Coin, CosmosMsg, CustomQuery, DepsMut, Env, MessageInfo, QuerierWrapper, Reply,
+    Response,
+};
+use cw_multi_test::{AppResponse, Contract, Executor};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+impl<T> Contract<T> for Cw1WhitelistContract<T>
+where
+    T: Clone + std::fmt::Debug + PartialEq + JsonSchema + DeserializeOwned,
+{
+    fn instantiate(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: Vec<u8>,
+    ) -> AnyResult<Response<T>> {
+        self.entry_instantiate(deps, env, info, &msg)
+            .map_err(Into::into)
+    }
+
+    fn execute(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: Vec<u8>,
+    ) -> AnyResult<Response<T>> {
+        self.entry_execute(deps, env, info, &msg)
+            .map_err(Into::into)
+    }
+
+    fn query(&self, deps: cosmwasm_std::Deps, env: Env, msg: Vec<u8>) -> AnyResult<Binary> {
+        self.entry_query(deps, env, &msg).map_err(Into::into)
+    }
+
+    fn sudo(&self, _deps: DepsMut, _env: Env, _msg: Vec<u8>) -> AnyResult<Response<T>> {
+        bail!("sudo not implemented for contract")
+    }
+
+    fn reply(&self, _deps: DepsMut, _env: Env, _msg: Reply) -> AnyResult<Response<T>> {
+        bail!("reply not implemented for contract")
+    }
+
+    fn migrate(&self, _deps: DepsMut, _env: Env, _msg: Vec<u8>) -> AnyResult<Response<T>> {
+        bail!("migrate not implemented for contract")
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct Cw1Executor<'a, App> {
+    addr: Addr,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+}
+
+impl<'a, App> Cw1Executor<'a, App> {
+    pub fn new(addr: Addr, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
+        Self {
+            addr,
+            app,
+            sender,
+            send_funds,
+        }
+    }
+
+    pub fn execute<C>(self, msgs: Vec<CosmosMsg<C>>) -> AnyResult<AppResponse>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema + Serialize + 'static,
+        App: Executor<C>,
+    {
+        self.app.execute_contract(
+            self.sender,
+            self.addr,
+            &Cw1ExecMsg::Execute { msgs },
+            self.send_funds,
+        )
+    }
+}
+
+#[derive(PartialEq, Debug)]
+#[must_use]
+pub struct WhitelistExecutor<'a, App> {
+    addr: Addr,
+    app: &'a mut App,
+    sender: Addr,
+    send_funds: &'a [Coin],
+}
+
+impl<'a, App> WhitelistExecutor<'a, App> {
+    pub fn new(addr: Addr, app: &'a mut App, sender: Addr, send_funds: &'a [Coin]) -> Self {
```

Comment:
This is a nice pattern. Always has the same `XyzExecutor.new()` args, then contract-specific function call.